### ISSUE TITLE
Series className is not applied to any part of the tooltip. [styled mode]

### ISSUE THREAD

**Issue by zlangner** [highcharts/highcharts#6448]
#### Expected behaviour
The series className should be applied to the appropriate items in the tool tip as well as the chart elements. Specifically the tspan element in the tooltip that has the "highcharts-color-0" class should also include what is specified by the series className.

#### Actual behaviour
The class is applied to the chart fine but NOT the tooltip.

#### Live demo with steps to reproduce
1. [Demo](http://jsfiddle.net/se7ta46c/)
2. Notice: the custom class "make-it-so" is being applied to the series and the legend which makes them appear red instead of the standard blue.
3. Hover on any bar and the tooltip still uses the standard blue. Using the Chrome dev tools I can see that the class is not applied to any part of the tool tip.

#### Affected browser(s)
All, the class isn't in the mark up.

**Comment by KacperMadej** [comment]
Hi,

Thank you for notifying us about the problem.
You could set `highcharts-color-0` and series and series tooltip will have the same colors. Demo: http://jsfiddle.net/se7ta46c/1/

Custom class is being applied for series only and tooltip is not a part of series - this why the style is not applied. I am not convinced that this is a bug - you are setting class name for series and not for tooltip. If you want to change color for series and tooltip for that series you could use `highcharts-color-0`.

**Comment by zlangner** [comment]
Yes, I know that for the fiddle provided I could just set the style up using 'highcharts-color-0'. However, in my real world example I have a page with dozens of charts each with multiple series and often times the same entity is represented in multiple charts. When this happens I want the series to use the same marker.symbol and be colored the same across charts to make the relationship more obvious. 

Here is an [updated fiddle](http://jsfiddle.net/se7ta46c/5/) showing 2 charts. Each has 3 series (which must be ordered alphabetically). I don't believe it is currently possible to color the symbol in the tooltip to match the charts and legends without 1) referencing the charts by id or 2) writing a custom tooltip. For my situation 1 is not possible as the chart ids are randomly generated and 2 seems excessive since the default tooltip is almost perfect. It just needs to apply the series's custom class to the marker being shown next to the series name. Please note, I am not suggesting the custom class be applied to the tooltip itself, but rather to elements within it. Specifically, the `tspan` elements that have `highcharts-color-{x}` classes.

**Comment by KacperMadej** [comment]
Thank you for providing more info.

As a workaround you could set globally `tooltip.pointFormat` and use a wrapper that will set border color for tooltip based on the` className` option of a series. Demo: http://jsfiddle.net/3nz7t205/

**Comment by TorsteinHonsi** [comment]
It sounds to me that your case can be resolved by using the [colorIndex] option: http://jsfiddle.net/se7ta46c/5/. It allows setting the color index on a series level.

Unfortunately it seems like that option hasn't been documented, it is only documented for the point level. I'll add it to the API right away.

**Comment by zhang-xinchao** [comment]
First, please correct me if I am wrong, the issue is not just that className is not applied in tooltip, in fact even if I manually construct the tooltip with html, the `class="{series.className}` is not honored. **It seems in styled mode the string interpolation does not work in attributes?**.

I would like to provide a different perspective regarding why I also believe the series className **should/is expected to** be applied to tooltip, especially in styled mode. In many areas, people would prefer deterministic styling/coloring of data series with specific meanings. For example, in finance, we have dedicated colors for _Buy_ and _Sell_, and we still prefer consistent global color allocation to _Countries_, _Industries_, or _Sectors_. For projects that are complex enough, where data is often load dynamically, we wouldn't be able to know what series, as well as their ordering, will be beforehand. The styled mode, if applied consistently to all components on the chart, could provide a solution to the problem as it effective allows data-driven style binding.

One of the major strength of highcharts in my view is that the library views chart as a holistic concept rather a collection of disjoint components. From my perspective, the visual components (series canvas, legend, tooltip) are just different views of the same configuration/data, hence if someone gives series an attribute, it is natural for that attribute to be reflected everywhere.

**Comment by TorsteinHonsi** [comment]
> Tried a few different things and the current workaround I have is to build a custom tooltip in html with class="{series.options.className}"

Yes that is the solution I would use, but I would hardly call it a _workaround_. It's good use of the API in my opinion. Simple demo: http://jsfiddle.net/highcharts/Lune2ok3/1/

**Comment by benelliott** [comment]
I think that the series class should be applied to the tooltip entry for the exact same reason it's applied to the legend item entries. The series is a more general concept than a specific line on the graph and pervades through all visual elements of the chart as @zhang-xinchao mentioned.

@TorsteinHonsi would you accept a PR that added this functionality?

**Comment by TorsteinHonsi** [comment]
Yes, let's see how it turns out 👍

**Comment by RandScullard** [comment]
This is a must for split tooltips. Like @zhang-xinchao, I have series with specific meaningful colors. I wanted to use a className for each series, but then my split tooltips are all the wrong color. It really seems like a bug that the series className is applied to legend items but not to split tooltips.

**Comment by benelliott** [comment]
Sorry - this must have dropped off my radar but I'll take another look at making a PR!
